soc/arm/nordic_nrf/nrf53/soc.c

```diff
@@ -137,12 +218,199 @@ bool z_arm_on_enter_cpu_idle(void)
 		suppress_message = true;
 	}
 #endif
+#if defined(CONFIG_SOC_NRF53_RTC_PRETICK) && defined(CONFIG_SOC_NRF5340_CPUNET)
+	if (ok_to_sleep) {
+		uint32_t rtc_ticks_to_next_event = 0U;
+
+		uint32_t rtc_counter = nrf_rtc_counter_get(NRF_RTC1);
+
+		barrier_dmem_fence_full();
+		bool rtc_scheduled = soc_rtc_ticks_to_next_event_get(NRF_RTC1, RTC1_PRETICK_SELECTED_CC_MASK, rtc_counter, &rtc_ticks_to_next_event);
+
+		uint32_t rtc0_ticks_to_next_event = 0U;
+		uint32_t rtc0_counter = nrf_rtc_counter_get(NRF_RTC0);
```

Comment:
I think that instead of reading rtc0 directly a better idea would be first calling `rtc_counter_diff_get` and then base everything on the acquired value of the RTC1 counter (so the RTC0 counter value would be simply RTC1 counter value + `diff` and the RTC0 is not read directly outside of `rtc_counter_diff_get`). There are lots of operation between reading the RTC1 and RTC0 counter values and there is no protection against RTC1 ticking in the mean time. This way the value RTC0 can already be ahead of RTC1 and CC events from RTC0 could be treated as being "earlier" although in reality they are later then the RTC1 events.

So my suggestion would be:
```
uint32_t diff = rtc_counter_diff_get(NRF_RTC0, NRR_RTC1);
uint32_t rtc_counter = nrf_rtc_counter_get(NRF_RTC1);
uint32_t rtc0_counter = rtc_counter + diff;
```

Reading RTC1 counter after diff also makes sure it has the newest possible value.

Reply:
I do not see a reason why you can't.

You want to know the difference to the nearest event of RTC0 and RTC1 from a _common_ point in time. Currently the reference time for RTC0 and RTC1 is different, and it shouldn't.

Reply:
(Of course earlier I meant `uint32_t rtc0_counter = (rtc_counter + diff) & NRF_RTC_COUNTER_MAX;`, I forgot about `& NRF_RTC_COUNTER_MAX`)

Reply:
Fixed, please have a look.